extra_data/tests/test_reader_mockdata.py

```diff
@@ -701,6 +701,19 @@ def test_open_run(mock_spb_raw_run, mock_spb_proc_run, tmpdir):
         for path in proc_paths:
             assert '/raw/' not in path
 
+        # All folders
+        run = open_run(proposal=2012, run=238, data='all')
+
+        for source, files in run._source_index.items():
+            for file in files:
+                if '/DET' in source:
+                    # AGIPD data is in proc.
+                    assert '/raw/' not in file.filename
+                else:
+                    # Non-AGIPD data is in raw.
+                    # (CAM, XGM)
+                    assert '/proc/' not in file.filename
```

Comment:
Any particular reason to check `not in` in both cases? I forget now why I did one `in` and one `not in` in the code just above.

Is it worth checking that both kinds of sources are actually there? At the moment, this would pass if there were no sources at all, I think.

Reply:
Ha, fair point with the sources check! But yes, I wanted to mirror your assertions.

Reply:
Added a test of sources for `open_run(data='all')`. Strictly speaking, the mock SPB run is a special case where `raw` contains all sources and `proc` only a select few, but this should neatly cover the two real world examples of

a) copy of all `raw` data in `proc` for traditiona detectors and
b) additional sources in `proc` for REMI data